docs/source/index.rst

```diff
@@ -1,20 +1,204 @@
+
+.. ISSUE: Tree doesn't include subsections for `self` in toctree.
+          https://github.com/sphinx-doc/sphinx/issues/2103
 
 .. toctree::
    :maxdepth: 2
-   :caption: Contents:
+   :hidden:
+
+   Home <self>
+
+.. EDUCATIONAL: The meta field below is for search engine optimization.
+.. meta::
+   :description: Turn complex GraphQL queries into optimized database queries.
+   :keywords: graphql compiler, database, orientdb, sql
+
+The GraphQL Compiler is a library that simplifies database querying and exploration by exposing one
+common query language for multiple database backends.  The query language is:
+
+.. EDUCATIONAL: The pattern below is what you would call a definition list in restructuredtext.
+   The "terms" get special rendering in the readthedocs html file.
+
+Written in valid GraphQL syntax
+   Since it uses GraphQL syntax, the user get access to the entire GraphQL ecosystem,
+   including the typeahead capabilities and query validation capabilities of `GraphiQL
+   <https://github.com/graphql/graphiql>`__, user friendly error messages from the
+   reference GraphQL python implementation, and more.
+
+Directly compiled to the target database language
+   By compiling instead of interpreting the query language, the compiler highly improves query
+   performance and empowers the user with the ability to write deep and complex queries.
+   Furthermore, by using schema information from the target database, the compiler is able to
+   extensively validate queries, often more so than the DB-API, (e.g. :code:`pymssql`).
+
+Designed for cross-database querying
+   Since the query language always has the same semantics regardless of the underlying database,
+   we have been able to build a :doc:`Schema Stitching <advanced_features/schema_stitching>` system
+   that allows for seamless cross-database querying.
+
+Getting Started
+---------------
+
+Generating the necessary schema info
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  TODO: Encapsulate all schema info in a SchemaInfo class.
+
+To use the GraphQL compiler the first thing one needs to do is to generate the schema info from the
+underlying database as in the example below. Even though the example below generates schema info
+from an OrientDB database, it is meant as a generic schema info generation example.
+See the target database homepage for schema generation instructions.
+
+.. code:: python
+
+    from graphql_compiler import (
+        get_graphql_schema_from_orientdb_schema_data
+    )
+    from graphql_compiler.schema_generation.orientdb.utils import ORIENTDB_SCHEMA_RECORDS_QUERY
+
+    client = your_function_that_returns_an_orientdb_client()
+    schema_records = client.command(ORIENTDB_SCHEMA_RECORDS_QUERY)
+    schema_data = [record.oRecordData for record in schema_records]
+    schema, type_equivalence_hints = get_graphql_schema_from_orientdb_schema_data(schema_data)
+
+At the core of generated schema info is the GraphQL :code:`schema`. The database might be
```

Comment:
I'm not sure it makes sense to decorate words as `code` if it's not actually code / an object name.

Reply:
schema is part of the code above.

Reply:
I agree with @MichaelaShtilmanMinkin, I think the overall paragraph could use some clarification:
- Saying the schema is at the core of the schema is ... kind of circular. Some clarification could help.
- The wording is also somewhat misleading, and the reader could believe that `schema` is a string with the contents starting on line 75, which is wrong in two ways -- it's not a string, and if it were, those would not be the (full) contents of it.
- `type_equivalence_hints` is assigned and named in the code, but then ignored altogether. At least mention where the curious reader can go to learn more about it.

Also, "GraphQL `schema`" just looks awkward from a purely stylistic angle, in my opinion.

Reply:
I addressed the comments. I didn't want to talk about schema printing here. I didn't want to expose the entire schema either. I left a TODO in the `Schema Types` file to talk about schema printing there.